controllers/designateapi_controller.go

```diff
@@ -760,7 +761,8 @@ func (r *DesignateAPIReconciler) reconcileNormal(ctx context.Context, instance *
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 {
+	if instance.Status.ReadyCount > 0 &&
```

Comment:
the ReadyCount > 0 statement could cause trouble during update. E.g. if there are 3 replicas requested and one is ready with the new image it does not mean that the other two replicas are running the new image as well.

Reply:
Yeah, I think we had  a similar conversation for `octavia-operator`. Fixing it and moving it to `instance.Status.ReadyCount == *instance.Spec.Replicas`, to make sure we report readiness only after the Pods have been rolled out.

Reply:
Done